internal/querynode/query_shard_service.go

```diff
@@ -0,0 +1,56 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package querynode
+
+import (
+	"context"
+	"sync"
+)
+
+type queryShardService struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	queryShardsMu sync.Mutex              // guards queryShards
+	queryShards   map[Channel]*queryShard // Virtual Channel -> *queryShard
+}
+
+func newQueryShardService(ctx context.Context) *queryShardService {
+	queryShardServiceCtx, queryShardServiceCancel := context.WithCancel(ctx)
+	qss := &queryShardService{
+		ctx:         queryShardServiceCtx,
+		cancel:      queryShardServiceCancel,
```

Comment:
NIT: Same context and cancel issue here.

Reply:
fixed in #16317